.github/workflows/build_gradle_sam.yml

```diff
@@ -0,0 +1,26 @@
+name: di-gradle-sam
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'aws-sam/*'
```

Comment:
This looks like it's a race condition.

When you change aws-sam, it will rebuild aws-sam and gradle-sam in parallel. gradle-sam will see the *old* aws-sam, if I'm not mistaken?

I wonder if we could trigger a build when the `aws-sam` package is published. Maybe we could use a `registry_package` event and a conditional like `if: github.event.package.name == 'aws-sam'`?

I'm no Actions expert though.

Reply:
Yep, this works way better, I didn't know `registry_package` was a event type! Looks like doing a `if: github.event.package.name == 'aws-sam'` is the only option to currently filter the package names though